routers/repo/download.go

```diff
@@ -21,16 +22,15 @@ import (
 
 // ServeData download file from io.Reader
 func ServeData(ctx *context.Context, name string, reader io.Reader) error {
-	buf := make([]byte, 1024)
-	n, err := reader.Read(buf)
+	content, err := ioutil.ReadAll(reader)
```

Comment:
Is there any limit on the size of the reader here?

Could this reader be for a file >100Mb, >1Gb?

You can't write code like this.

Reply:
Yes, this reader is used for the /raw route. So, it can be used for files hosted in a git repo (of arbitrary sizes naturally).

I needed to do it like this, to get the content length before the call to ctx.Resp.Write(content). When I do it after the call to io.Copy(ctx.Resp, reader), it is not taken into account and the Content-Length header is not added to the response.

Reply:
You could look at what is sending the `io.Reader`. Presumably it has a file handle and may be able to pass along the content length.

Reply:
Another solution would be to pass a an int64 size argument to ServeData().

This function is called 3 times : 
- routers/repo/attachment.go:GetAttachment() function : with Attachment object which has a Size field.
- routers/repo/download.go:ServeBlob() function : with a git.Blob object it also has a Size field.
- routers/repo/download.go:ServeBlobOrLFS() function : with a git.Blob or LFSMetaObject, and both have a Size field.

This makes it a lot easier, without changing the original call to io.Copy(). and it is more secure.